src/render_apollo_test.js

```diff
@@ -96,47 +62,79 @@ describe('render apollo', () => {
             },
         },
     };
+    let mockScope;
 
-    it('should correctly render a simple apollo react component', (done) => {
-        mockScope.post("/graphql").reply(200, validGraphQLResponse);
-        render(packages,
-            "./javascript/coach-package/valid.jsx",
-            {},
-            {
-                ApolloNetwork: {
-                    url: "https://www.ka.org/graphql",
-                },
-            }
-        ).then(actual => {
-            assert.deepEqual(expectedBase, actual);
-            done();
-        }).catch(done);
+    before(() => {
+        nock.disableNetConnect();
+        nock.enableNetConnect('127.0.0.1');
     });
 
-    it('should handle variables', (done) => {
+    beforeEach(() => {
+        mockScope = nock('https://www.ka.org');
+        render.resetGlobals();
+        cache.init(1000000);
+    });
+
+    afterEach(() => {
+        nock.cleanAll();
+        render.resetGlobals();
+        cache.destroy();
+    });
+
+    it('should correctly render a simple apollo react component', async () => {
+        // Arrange
         mockScope.post("/graphql").reply(200, validGraphQLResponse);
-        render(packages,
-            "./javascript/coach-package/variables.jsx",
-            {
-                name: "Test Class 1",
+        const packages = loadPackages([
+            "webpacked/common/1.js",
+            "webpacked/common/2.js",
+            "webpacked/common/3.js",
+            "webpacked/apollo/simple/entry.js",
+        ]);
+
+        // Act
+        const result = await render(packages, {}, {
+            ApolloNetwork: {
+                url: "https://www.ka.org/graphql",
             },
-            {
-                ApolloNetwork: {
-                    url: "https://www.ka.org/graphql",
-                },
-            }
-        ).then(actual => {
-            assert.deepEqual(expectedBase, actual);
-            done();
-        }).catch(done);
+        });
+
+        // Assert
+        assert.deepEqual(result, expectedBase);
     });
 
-    const headers = {
-        "Cookie": "foo=1234",
-        "X-KA-Fkey": "secretkey",
-    };
+    it('should handle variables', async () => {
+        // Arrange
+        mockScope.post("/graphql").reply(200, validGraphQLResponse);
+        const packages = loadPackages([
+            "webpacked/common/1.js",
+            "webpacked/common/2.js",
+            "webpacked/common/3.js",
+            "webpacked/apollo/variables/entry.js",
+        ]);
+
+        // Act
+        const result = await render(packages, {name: "Test Class 1"}, {
+            ApolloNetwork: {
+                url: "https://www.ka.org/graphql",
+            },
+        });
+
+        // Assert
+        assert.deepEqual(result, expectedBase);
```

Comment:
Not sure I understand the question exactly, so I'll answer the two things I think it might be.

1. The prop is used in the component to render the outcome. If line 116 were changed to change the `name` prop, the test would fail as it would no longer render what the base expects.
2. The prop is used to make a gql query and therefore the expected data request won't be the same if we changed the prop